sql/views.sql

```diff
@@ -326,6 +327,10 @@ SELECT
     END
     AS err_message
 FROM
-    _timescaledb_internal.job_errors;
-
+    _timescaledb_internal.job_errors
+JOIN
+    _timescaledb_config.bgw_job ON (bgw_job.id = job_errors.job_id)
+WHERE
+    pg_catalog.pg_has_role(current_user, owner, 'MEMBER'::text);
```

Comment:
I've been thinking about this one, we may want to do something like this: (PG14 and above only):

The `LEFT JOIN`, as some older jobs that have been removed from `bgw_job` may still have logs in the `job_errors` table, and they should somehow be made available for inspection?

However, if the `LEFT JOIN` results in a `null` `owner`, we have to ensure we have a second where clause that would allow a (super)(admin) user to see the logs?

```suggestion
LEFT JOIN
    _timescaledb_config.bgw_job ON (bgw_job.id = job_errors.job_id)
WHERE
    pg_catalog.pg_has_role(current_user, 'pg_database_owner', 'MEMBER'::text)
    OR pg_catalog.pg_has_role(current_user, owner, 'MEMBER'::text);
```

Or this:

```suggestion
LEFT JOIN
    _timescaledb_config.bgw_job ON (bgw_job.id = job_errors.job_id)
WHERE
    pg_catalog.pg_has_role(current_user, 'timescale_predefined_owner_role', 'MEMBER'::text)
    OR pg_catalog.pg_has_role(current_user, owner, 'MEMBER'::text);
```

Reply:
The above is however also not strictly required to change, as we can say that `_timescaledb_internal.job_errors` is available for database owners and superusers.

Reply:
Hmmm... even though it is possible to query the `_timescaledb_internal.job_errors` table directly as a superuser even though it contains NULL entries, it seems to be more complicated than necessary. It is probably better, from a usability perspective, to allow entries with NULL owner to be seen by a superuser and database owner through the view, so I think this change is good.

Using the existing `pg_database_owner` is good, but we need to think about how to handle the case where that pre-defined role does not exist. Simplest would probably be to extract the owner of the database using `current_database` and dig through PostgreSQL tables.